packages/flutter/lib/src/rendering/object.dart

```diff
@@ -210,6 +231,13 @@ class PaintingContext extends ClipContext {
     _containerLayer.append(_currentLayer);
   }
 
+  /// Returns a layer containing all content painted.
+  @protected
+  OffsetLayer toLayer() {
+    _stopRecordingIfNeeded();
+    return _containerLayer;
```

Comment:
toLayer being destructive (it stops recording) seems unintuitive. We should definitely document all the side-effects, but also, we should probably name this something like "build" (like in ParagraphBuilder and SceneBuilder).

Reply:
Good idea! Switched to `build()` and documented side effects.